CheckedExceptions/CheckedExceptionsAnalyzer.cs

```diff
@@ -1364,72 +1269,57 @@ private bool IsExceptionDeclaredInMember(SyntaxNodeAnalysisContext context, Synt
+                // In that case, the accessors are analyzed separately
+                if ((propertySymbol?.GetMethod is not null && propertySymbol?.SetMethod is not null)
+                    || (propertySymbol?.GetMethod is null && propertySymbol?.SetMethod is not null))
+                {
+                    return false;
+                }
 
-                    // Don't continue with the analysis if it's a full property with accessors
-                    // In that case, the accessors are analyzed separately
-                    if ((propertySymbol?.GetMethod is not null && propertySymbol?.SetMethod is not null)
-                        || (propertySymbol?.GetMethod is null && propertySymbol?.SetMethod is not null))
+                var propertySyntaxRef = propertySymbol?.DeclaringSyntaxReferences.FirstOrDefault();
+                if (propertySyntaxRef is not null && propertySyntaxRef.GetSyntax() is PropertyDeclarationSyntax basePropertyDeclaration)
+                {
+                    if (basePropertyDeclaration.ExpressionBody is null)
                     {
                         return false;
                     }
+                }
+                
+                symbol = propertySymbol;
+            }
+            else symbol = ancestor switch
+            {
+                BaseMethodDeclarationSyntax methodDeclaration => context.SemanticModel.GetDeclaredSymbol(methodDeclaration),
+                AccessorDeclarationSyntax accessorDeclaration => context.SemanticModel.GetDeclaredSymbol(accessorDeclaration),
+                LocalFunctionStatementSyntax localFunction => context.SemanticModel.GetDeclaredSymbol(localFunction),
+                AnonymousFunctionExpressionSyntax anonymousFunction => context.SemanticModel.GetSymbolInfo(anonymousFunction).Symbol as IMethodSymbol,
+                _ => null,
+            };
 
-                    var propertySyntaxRef = propertySymbol?.DeclaringSyntaxReferences.FirstOrDefault();
-                    if (propertySyntaxRef is not null && propertySyntaxRef.GetSyntax() is PropertyDeclarationSyntax basePropertyDeclaration)
-                    {
-                        if (basePropertyDeclaration.ExpressionBody is null)
-                        {
-                            return false;
-                        }
-                    }
-
-                    symbol = propertySymbol;
-                    break;
-
-                case AccessorDeclarationSyntax accessorDeclaration:
-                    symbol = context.SemanticModel.GetDeclaredSymbol(accessorDeclaration);
-                    break;
-
-                case LocalFunctionStatementSyntax localFunction:
-                    symbol = context.SemanticModel.GetDeclaredSymbol(localFunction);
-                    break;
-
-                case AnonymousFunctionExpressionSyntax anonymousFunction:
-                    symbol = context.SemanticModel.GetSymbolInfo(anonymousFunction).Symbol as IMethodSymbol;
```

Comment:
@marinasundstrom are you intentionally casting to `IMethodSymbol` here or did you forget to remove this?